### ISSUE TITLE
Error encountered when opening  drop plan settings

### ISSUE THREAD

**Issue by Alfadbhq** [yanivsegev/VSTS-DropPlan#115]
with the latest updates, we are unable to open drop plan settings and stalled in "Please wait while loading data..."  Framework loaded.
![MicrosoftTeams-image](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/af55f748-d0db-4dc7-a888-29a6f2b33f51)

**Comment by sgtwilko** [comment]
Do you have an iteration configured?

**Comment by Alfadbhq** [comment]
Hi @sgtwilko,

Thanks for the quick reply.  We have configured iteration for the entire team (see attached). When we first encountered this, we just go to settings and update, apparently for this project we cannot do that and stalled when loading.
![Project Iterations](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/e4da72ae-5a31-47ad-9823-2696dbfa7e29)
![Droplan Sprint](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/9943a091-4111-4835-a8ae-e21a36d327f5)

**Comment by sgtwilko** [comment]
Hiya @Alfadbhq 
woah, that's Drop Plan is not pretty!

How many siblings do those tasks have?  
I counted one with 103 dots. indicating 103 tasks in a story.

Can you also send a screenshot showing the warnings for one or two of those tasks please.

**Comment by Alfadbhq** [comment]
Hello again @sgtwilko

104 to be exact :) anyways all the warnings are  "Testing starting before Development has finished!." For more than 30 projects that uses this app in ADO, these happens on these project only. Do you think the number of tasks per user story has something do with it
![Warnings](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/bbb1a145-111b-463a-84f5-c9655ae580d2)
?

**Comment by sgtwilko** [comment]
Hi @Alfadbhq ,
Couple of questions.

Are you a member of the team?
Are you a member of multiple teams in the project?
What's your Azure Devops licence level (basic, stakeholder, etc)
Do you have permission to view the Project Process?

**Comment by sgtwilko** [comment]
Hiya @Alfadbhq ,

The latest version has a fix for your large number of tasks.
Also a little debug in the settings page, so a fresh screenshot of that would be handy.

(Would also be nice to see how the drop plan looks with the large number of tasks fix)

**Comment by Alfadbhq** [comment]
Hi @sgtwilko,

Apologies for late reply. 

Will try to answer your question one by one in summary.

Our team setup is (see attached) for the whole of our Division is setup that way. No order enforced.

![DropPlan Team Settings](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/59f1046a-ccb7-4bd7-9ce8-7f768846f9da).

I oversee all the project settings even am not part of the team with a Basic + Test Plan license. All teams using the sprint drop plan are with an active iteration and with Basic or Basic + Test plans license as well.

The latest fixed did not resolved the issue, still cannot modify the drop plan settings of the affected project. (see latest screenshots). Hence, discuss the issue with the scrum master to break the user story into two, just to limit large tasks under one user story. Will send another update once breaking user story is done.

![dropplan settings](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/fc7b9dd6-94c7-4d36-b6f6-847a28939a8e)

![Droplan 021224](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/d24534af-64e0-4aab-af5e-007f16a7ff26)

**Comment by sgtwilko** [comment]
Hiya @Alfadbhq 
The last screenshot looks a lot better to me, you no longer have things cascading down the screen.

The first screenshot shows the settings working correctly, was that from someone who is a team member?

Was the screenshot of the settings working on the Same project as the final screenshot?
Those settings are per project, so if you want to disable the warning you would need just once on each project.

I think that the size of the stories will Not affect the settings.  You should not need to break them down to get the settings to work.
I suspect that DevOps is not returning the team id when you access the settings page, hence why I was asking if you're part of the team.
I'll see if there's another way to access the information.

**Comment by sgtwilko** [comment]
Hi @Alfadbhq ,

Can you confirm that the changes made are now no longer showing the warning on each task?
Can you also confirm if you are the only person unable to see the settings page.

Thanks

**Comment by Alfadbhq** [comment]
Hi @sgtwilko,

Thanks for the update. Apparently it still showing, I already included myself as part of the team and project administrator as well just to check if there's change in behavior but it's still the same.  attached is the latest screenshot.

Rest of the team are unable to see the setting page as well.

![Sprint](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/859c6396-083d-4cac-a5ff-587fe05d90e2)
![Drop Plan Settings](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/fbbb258b-e9a9-4c0a-887d-6c230c6befdc)

Regards :)

**Comment by sgtwilko** [comment]
Hi @Alfadbhq ,
How did you get that screenshot of the settings page working from the 12th?
Was that you or a different user?

**Comment by Alfadbhq** [comment]
Hi @sgtwilko,

The screenshot came from me, since i have the access to all 30 projects and teams that are using this extension for daily scrum meeting, however, only one project is having issues when loading drop plan settings.

**Comment by sgtwilko** [comment]
@Alfadbhq 

I've replicated the issue.

Please can you check the team settings for the project that is failing.

Do you have a blank backlog iteration like the following image:
![image](https://github.com/yanivsegev/VSTS-DropPlan/assets/658876/5d641b9e-11a5-466a-b8ca-5d42f4260749)

**Comment by Alfadbhq** [comment]
Hi @sgtwilko,

Here's the snapshot of teams iteration, these includes past, current and future iteration. 

![opds1_1](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/8d2020e4-f4e8-42ae-b4fe-7e657b00643f)
![opds1-2](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/61007342-68d0-4108-b44c-885b2a41ee24)
![opds1-3](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/99d0f4bc-4f91-45f0-8ea4-8fda92518b65)
![opds1-4](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/840e70f8-4a7d-4172-a597-e7958898c0f0)

 Regards

**Comment by sgtwilko** [comment]
@Alfadbhq 
More logging was added to the settings page in 2.2.0.
Please can you check you are on at least 2.2.0 and then can you retry the failing page and let me have the details from the console please.

Thanks

**Comment by Alfadbhq** [comment]
Hi @sgtwilko,

We are using 2.2.1, and we encountered error settings when opening erroneous project, hence rest of the projects are working as expected.
![SDP ops error](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/8bf7ca3e-3a44-46ae-a0e3-e868e5d7f0ed)

![SDP version](https://github.com/yanivsegev/VSTS-DropPlan/assets/156981307/5f1eeea0-8591-4256-b750-78688415c831)
